packages/@webex/internal-plugin-metrics/src/prelogin-metrics-batcher.ts

```diff
@@ -0,0 +1,97 @@
+/* eslint-disable class-methods-use-this */
+/* eslint-disable valid-jsdoc */
+
+import {uniqueId} from 'lodash';
+import Batcher from './batcher';
+import {prepareDiagnosticMetricItem} from './call-diagnostic/call-diagnostic-metrics.util';
+import {generateCommonErrorMetadata} from './utils';
+
+const PRE_LOGIN_METRICS_IDENTIFIER = 'Pre Login Metrics -->';
+
+const PreLoginMetricsBatcher = Batcher.extend({
```

Comment:
Why is this batcher created when `call-diagnostics-metrics-batcher` exists? Is there a reason why we did not reuse the existing batcher, or expand on the functionality of the existing batcher?

Reply:
they're hitting different APIs which is why i had it separate.
This will be refactored once Metrics MVP comes into play.. metrics a supports multiple types of metrics, and we need to prepare the events in different ways. In this case It's the prepare item and request match the call-diagnostic-metrics-batcher, but it won't always do. Nonetheless could still be merged into 1 batcher, but I didn't want to refactor too much code now, when I know this will be refactored in the near future.